libs/sdk-ui-kit/src/Menu/tests/Menu.test.tsx

```diff
@@ -21,126 +17,97 @@ const MenuContent = (props: React.HTMLAttributes<HTMLElement>) => {
 
 describe("Menu renderer", () => {
     it("should render the toggler", () => {
-        const wrapper = mount(
-            <Menu toggler={<Toggler />}>
-                <MenuContent />
-            </Menu>,
-        );
+        renderComponent();
 
-        expect(wrapper.contains(<Toggler />)).toBeTruthy();
-        wrapper.unmount();
+        expect(screen.getByText("toggler")).toBeInTheDocument();
     });
 
     it("should render the menu content to body", () => {
-        const wrapper = mount(
-            <Menu toggler={<Toggler />} opened={true}>
-                <MenuContent />
-            </Menu>,
-        );
+        renderComponent({ menu: { opened: true } });
 
-        expect(isContentRenderedInBody()).toBeTruthy();
-        wrapper.unmount();
+        expect(isContentRenderedInBody()).toBeInTheDocument();
+        expect(screen.getByText("1")).toBeInTheDocument();
     });
 
     it("should render the menu content to portal target", () => {
         const portalTarget = document.createElement("section");
-        const wrapper = mount(
-            <Menu toggler={<Toggler />} portalTarget={portalTarget} opened={true}>
-                <MenuContent className="content-portaled" />
-            </Menu>,
-        );
-
+        const props = {
+            menu: {
+                opened: true,
+                portalTarget,
+            },
+            menuContent: {
+                className: "content-portaled",
+            },
+        };
+        renderComponent(props);
+
+        expect(isContentRenderedInBody()).not.toBeInTheDocument();
         expect(portalTarget.querySelector(".content-portaled")).toBeTruthy();
-        wrapper.unmount();
     });
 });
 
 describe("Menu toggling", () => {
-    it("should toggle menu when toggler is clicked", () => {
-        const wrapper = mount(
-            <Menu toggler={<Toggler />}>
-                <MenuContent />
-            </Menu>,
-        );
+    it("should toggle menu when toggler is clicked", async () => {
+        renderComponent();
 
-        expect(isContentRenderedInBody()).toBeFalsy();
-        wrapper.find(Toggler).simulate("mousedown");
-        wrapper.find(Toggler).simulate("click");
-        expect(isContentRenderedInBody()).toBeTruthy();
-        wrapper.find(Toggler).simulate("mousedown");
-        wrapper.find(Toggler).simulate("click");
-        expect(isContentRenderedInBody()).toBeFalsy();
+        expect(isContentRenderedInBody()).not.toBeInTheDocument();
 
-        wrapper.unmount();
-    });
+        await userEvent.click(screen.getByText("toggler"));
 
-    it("should close when we click outside of menu", () => {
-        const outsideElement = document.createElement("button");
-        document.body.appendChild(outsideElement);
-
-        const wrapper = mount(
-            <Menu toggler={<Toggler />}>
-                <MenuContent />
-            </Menu>,
-        );
+        expect(screen.getByText("1")).toBeInTheDocument();
```

Comment:
Since there is a user's click we should use `find*` or `waitFor` to make sure we receive the element after the click

Reply:
Added `waitFor`.